gns3server/controller/project.py

```diff
@@ -1237,3 +1247,66 @@ def __json__(self):
 
     def __repr__(self):
         return "<gns3server.controller.Project {} {}>".format(self._name, self._id)
+
+    async def _duplicate_fast(self, name=None, location=None, reset_mac_addresses=True):
+        # remote replication is not supported
+        if not sys.platform.startswith("linux") and not sys.platform.startswith("win"):
+            return None
+        for compute in self.computes:
+            if compute.id != "local":
+                log.warning("Duplicate fast not support remote compute: '{}'".format(compute.id))
+                return None
+        # work dir
+        p_work = pathlib.Path(location or self.path).parent.absolute()
+        t0 = time.time()
+        new_project_id = str(uuid.uuid4())
+        new_project_path = p_work.joinpath(new_project_id)
+        # copy dir
+        scripts_path = os.path.join(pathlib.Path(__file__).resolve().parent.parent.parent, 'scripts')
+        process = await asyncio.create_subprocess_exec('python', os.path.join(scripts_path, 'copy_tree.py'), '--src',
+                                                       self.path, '--dst',
+                                                       new_project_path.as_posix())
```

Comment:
Why do you call an external Python script to copy the tree and not shutil.copytree() directly?

Reply:
Indeed, we can use shutil.copytree(), but it will cause the entire program request to be blocked. For example, the page cannot be refreshed in time. So, I think the external Python script may be a feasible plan.

If you have a better plan, that would be better.

Reply:
I thought about it, it's my business requirements
. If you don't mind requesting blocking, I can change the code, using shutil.copytree() is also no problem.

Reply:
Thank you, I'll try it. If it works, I'll submit it again.

Reply:
It works. I pushed it again. Thanks : )